pkg/util/pki/csr_test.go

```diff
@@ -603,7 +678,7 @@ func TestGenerateCSR(t *testing.T) {
 			literalCertificateSubjectFeatureEnabled: true,
 		},
 		{
-			name:                                    "Error on generating CSR from certificate without CommonName in LiteralSubject, uri names, email address, or ip addresses",
+			name:                                    "Error on generating CSR from certificate without CommonName in LiteralSubject, uri names, email address, ip addresses or otherName set",
 			crt:                                     &cmapi.Certificate{Spec: cmapi.CertificateSpec{LiteralSubject: "O=EmptyOrg"}},
 			wantErr:                                 true,
 			literalCertificateSubjectFeatureEnabled: true,
```

Comment:
In the `t.Run` function below, add the following lines to make sure the CertificateRequest object is actually valid:

```go
	if err != nil {
		return
	}

	pk, err := GenerateRSAPrivateKey(2048)
	if err != nil {
		t.Fatal(err)
	}

	csrDER, err := EncodeCSR(got, pk)
	if err != nil {
		t.Fatal(err)
	}

	_, err = x509.ParseCertificateRequest(csrDER)
	if err != nil {
		t.Fatal(err)
	}
```